src/Dumplie/Application/Metadata/HydrationData.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Metadata;
+
+interface HydrationData
+{
+    /**
+     * @param Storage $storage
+     *
+     * @return array
+     */
+    public function produce(Storage $storage) : array;
```

Comment:
With current solution, you can define whole metadata schema without instantiating storage.